dashboard/src/components/CollapsingMenu/CollapsingMenu.js

```diff
@@ -0,0 +1,81 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Hidden from "@material-ui/core/Hidden";
+import Menu from "@material-ui/core/Menu";
+import ButtonSet from "/components/ButtonSet";
+import VerticalDisclosureButton from "/components/VerticalDisclosureButton";
+
+let id = 0;
+const getNextId = () => {
+  id += 1;
+  return id;
+};
+
+const breakpoints = ["xs", "sm", "md", "lg", "xl"];
+
+class CollapsingMenu extends React.PureComponent {
+  static propTypes = {
+    children: PropTypes.node.isRequired,
+    breakpoint: PropTypes.oneOf(breakpoints.slice(1)),
+  };
+
+  static defaultProps = {
+    breakpoint: "sm",
+  };
+
+  constructor(props) {
+    super(props);
+    this._id = getNextId();
+  }
+
+  state = {
+    anchorEl: null,
+  };
+
+  render() {
+    const { breakpoint, children } = this.props;
+
+    const buttons = React.Children.map(children, child =>
+      React.cloneElement(child, { renderAs: "button" }),
+    );
+    const menuId = `collapsed-menu-${this._id}`;
+    const menuItems = React.Children.map(children, child =>
```

Comment:
I'm on the fence about this approach. I dislike seeing the coupling to each button child, this might not work if a button is a wrapped component.|